propagator/propagator.go

```diff
@@ -50,6 +54,32 @@ func (e errInvalidHeader) Error() string {
 	return fmt.Sprintf("invalid header %s", e.header)
 }
 
+// CloudTraceOneWayPropagator will extract Trace IDs from the
+// x-cloud-trace-context header, and inject the w3c-standard 'traceparent'
+// header.
+type CloudTraceOneWayPropagator struct {
+	propagation.TraceContext
+}
+
+func (p CloudTraceOneWayPropagator) Extract(ctx context.Context, carrier propagation.TextMapCarrier) context.Context {
+	header := carrier.Get(TraceContextHeaderName)
+	if header == "" {
+		return ctx
+	}
+	sc, err := spanContextFromXCTCHeader(header)
+	if err != nil {
+		log.Printf("CloudTraceFormatPropagator: %v", err)
+		return ctx
+	}
+	return trace.ContextWithRemoteSpanContext(ctx, sc)
+}
+
+func (p CloudTraceOneWayPropagator) Fields() []string {
+	return []string{TraceContextHeaderName, TraceparentHeaderName}
```

Comment:
```suggestion
	return []string{traceparentHeaderName}
```
Based on [the docs](https://github.com/open-telemetry/opentelemetry-go/blob/main/propagation/propagation.go#L108), fields should only include those set during Inject().  Also, make it unexported.